src/Aspire.Hosting.Keycloak/KeycloakResouceBuilderExtensions.cs

```diff
@@ -0,0 +1,129 @@
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// The container is based on the quay.io/keycloak/keycloak container image.
+    /// The default tag is 24.0. The container exposes port 8080 by default.
+    /// </remarks>
+    /// <example>
+    /// <code lang="csharp">
+    /// var keycloak = builder.AddKeycloak("keycloak");
+    ///
+    /// var myService = builder.AddProject&lt;Projects.MyService&lt;()
+    ///                        .WithReference(keycloak);
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<KeycloakResource> AddKeycloak(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? adminUsername = null,
+        IResourceBuilder<ParameterResource>? adminPassword = null)
+    {
+        var passwordParameter = adminPassword?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
+
+        var resource = new KeycloakResource(name, adminUsername?.Resource, passwordParameter);
+
+        var keycloak = builder
+            .AddResource(resource)
+            .WithImage(KeycloakContainerImageTags.Image)
+            .WithImageRegistry(KeycloakContainerImageTags.Registry)
+            .WithImageTag(KeycloakContainerImageTags.Tag)
+            .WithHttpEndpoint(port: port, targetPort: DefaultContainerPort)
+            .WithEnvironment(context =>
+            {
+                context.EnvironmentVariables[AdminEnvVarName] = resource.AdminReference;
+                context.EnvironmentVariables[AdminPasswordEnvVarName] = resource.AdminPasswordParameter;
+            });
+
+        if (builder.ExecutionContext.IsRunMode)
+        {
+            keycloak.WithArgs("start-dev");
+        }
+        else
+        {
+            keycloak.WithArgs("start");
+        }
+
+        keycloak.WithArgs("--import-realm");
+
+        return keycloak;
+    }
+
+    /// <summary>
+    /// Adds a named volume for the data folder to a Keycloak container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only volume.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// The volume is mounted at /opt/keycloak/data in the container.
+    /// </remarks>
+    /// <example>
+    /// <code lang="csharp">
+    /// var keycloak = builder.AddKeycloak("keycloak")
+    ///                       .WithDataVolume();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<KeycloakResource> WithDataVolume(this IResourceBuilder<KeycloakResource> builder, string? name = null, bool isReadOnly = false)
```

Comment:
What scenario would a readonly data volume support?

Reply:
None that I can think of. Removed this parameter.